File: Part2-Regression/1SimpleLinearRegression/simple_linear_regression.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

# Splitting the dataset into the Training set and Test set
# install.packages('caTools')
library(caTools)
set.seed(123)

# 20 observations in the training set, 20 in test set
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Fitting simple linear regression to the training set
# @param formula (salary proportional to yrs exp)
# @param data (training set to derive fit)
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)

# Predicting Test set results
y_pred = predict(regressor, newdata = test_set)

# Visualizing Training set results
# install.packages('ggplot2')
# automatically include ggplot2 library
library(ggplot2)

ggplot() + 
  # plots the points from the training set w/ respective axes
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
            color = 'red') +
  # plot the regression line
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs. Experience (Training Set)') +
  # Label x axis
  xlab('Years of Experience') +
  ylab('Salary')


# Visualizing Test set results
ggplot() + 
  # plots the points from the test set w/ respective axes 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
            color = 'red') +
  # plot the same regression line regression line
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            color = 'blue') +
  ggtitle('Salary vs. Experience (Test Set)') +
  # Label x axis
  xlab('Years of Experience') +
  ylab('Salary')

File: Part2-Regression/1SimpleLinearRegression/simple_linear_regression.py
# Simple Linear Regression

# Importing the libraries
import sys
sys.path.append('/usr/local/lib/python3.6/site-packages')
import numpy as np
import matplotlib
matplotlib.use('TkAgg')
import matplotlib.pyplot as plt
import pandas as pd
import sklearn

# Importing the dataset
dataset = pd.read_csv('/Users/michaelchen/Desktop/MachineLearningA-ZTemplateFolder/Part2-Regression/1SimpleLinearRegression/Salary_Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

# Splitting the dataset into the Training set and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_y = StandardScaler()
y_train = sc_y.fit_transform(y_train)"""


# ===========================================================================


# Fitting Simple Linear Regression to the Training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
# fitting the regressor to the training set
regressor.fit(X_train, y_train)

# Predicting Test set results
# y_pred is the vector of predictions 
y_pred = regressor.predict(X_test)

# Visualizing the Training set results
plt.scatter(X_train, y_train, color = 'red') # plot the training coords
plt.plot(X_train, regressor.predict(X_train), color = 'blue') # predicted salaries from training set
plt.title('Salary vs. Experience (Training Set)') 
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visualizing the Test set results
plt.scatter(X_test, y_test, color = 'red') # plot the test set coords
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs. Experience (Test Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

